### PR TITLE
Make release_linux.sh configurable for python version

### LINKED ISSUE #1179: Make release_linux.sh configurable for python version
Currently our release for linux works quite differently than MacOS / Windows. All python versions are built serially within a docker container:
https://github.com/tensorflow/addons/blob/master/tools/ci_build/builds/release_linux.sh#L36

We should instead pass in an argument for which python version to build with. This would make it so our CI tests can build a single wheel file during the build check:
https://github.com/tensorflow/addons/blob/master/.github/workflows/release.yml#L32-L46

Currently the test times are:
- MacOS: ~9min
- Windows: ~14min
- Linux: ~23min

The current system also hardcodes the python versions in this script and is a maintenance burden when we update workflow files in the future.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Fix #1179

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,18 +32,23 @@ jobs:
...
         run: echo "::set-env name=NIGHTLY_FLAG::--nightly"
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v1
         with:
-          python-version: 3.5
+          python-version: ${{ matrix.python-version }}
       - name: Build manylinux2010 wheels
+        env:
+          PY_VERSION: ${{ matrix.python-version }}
         run: |
```
Comment (seanpmorgan) [inline .github/workflows/release.yml]: Can we add an if statement similar to the others:
```
 if: github.event_name == 'schedule' || github.event_name == 'release' || matrix.python-version == '3.5'
```
Probably best to do this after we confirm that we're correctly building 3.5/3.6/3.7

File: tools/ci_build/builds/release_linux.sh [hunk 2]
```diff
@@ -29,48 +29,22 @@ export CUDA_TOOLKIT_PATH="/usr/local/cuda"
...
-done
-
-# Clean up
-rm get-pip.py
+ln -sf $(which python$1) /usr/bin/python
+python -m pip install --upgrade pip
+python -m pip install --upgrade setuptools
+
+#Link TF dependency
+python ./configure.py --quiet
```
[the remark this answers is not in the document]
Comment (gabrieldemarmiesse) [reply tools/ci_build/builds/release_linux.sh]: Sure thing, I see the artifacts are wrong, but I don't understand why. Isn't a symlink enough?

File: tools/ci_build/builds/release_linux.sh [hunk 3]
```diff
@@ -29,48 +29,23 @@ export CUDA_TOOLKIT_PATH="/usr/local/cuda"
...
-
-# Clean up
-rm get-pip.py
+ln -sf $(which python$1) /usr/bin/python
+ln -sf $(which python$1) /usr/bin/python3
+python -m pip install --upgrade pip
+python -m pip install --upgrade setuptools
+
+#Link TF dependency
+python ./configure.py --quiet
```
Comment (seanpmorgan) [inline tools/ci_build/builds/release_linux.sh]:
```suggestion
ln -sf $(which python$1) /usr/bin/python3
python3 -m pip install --upgrade pip
python3 -m pip install --upgrade setuptools

#Link TF dependency
python3 ./configure.py --quiet
```

File: tools/ci_build/builds/release_linux.sh [hunk 4]
```diff
@@ -29,48 +29,22 @@ export CUDA_TOOLKIT_PATH="/usr/local/cuda"
...
-done
-
-# Clean up
-rm get-pip.py
+ln -sf $(which python$1) /usr/bin/python3
+python3 -m pip install --upgrade pip
+python3 -m pip install --upgrade setuptools
+
+#Link TF dependency
+python3 ./configure.py --quiet
```
Comment (seanpmorgan) [inline tools/ci_build/builds/release_linux.sh]:
```suggestion
python3 --version
python3 ./configure.py
```

Comment (seanpmorgan) [reply tools/ci_build/builds/release_linux.sh]: Debugging why that fails

**Comment by gabrieldemarmiesse** [comment]
https://stackoverflow.com/questions/44967202/pip-is-showing-error-lsb-release-a-returned-non-zero-exit-status-1/44967506 May solve the issue

**Pull request merged**